dartagnan/src/main/java/com/dat3m/dartagnan/expression/processing/ExprSimplifier.java

```diff
@@ -23,8 +25,39 @@ public ExprSimplifier(boolean aggressive) {
         this.aggressive = aggressive;
     }
 
+    // Tries to perform a general rewriting of a binary expression.
+    // For now, it only rewrites "ITE(cond, x, y) op z" to "ITE(cond, x op z, y op z)".
+    // Returns NULL, if no rewriting is applicable.
+    private Expression tryGeneralRewrite(BinaryExpression expression) {
+        final Expression left = expression.getLeft().accept(this);
+        final Expression right = expression.getRight().accept(this);
+        if (!(left instanceof ITEExpr) && !(right instanceof ITEExpr)
+                // We don't rewrite "ITE op ITE" for now
+                || (left instanceof ITEExpr && right instanceof ITEExpr)) {
+            return null;
+        }
```

Comment:
isn't the check simple if you check 
```
if one of them is ITE, 
  then apply the transformation
  else return null
```
?

You might even (I am not completely sure) use `if left instanceof ITEExpr ite` and avoid the explicit cast

Reply:
I apply this transformation only if exactly one of the expressions is an ITE but not both.
So I have to capture the cases `ITE op x` and `x op ITE` and both are different (because op my not be commutative)